write body for run_knob_identification

swimj opened the pull request:
I found and used the preprocessing.Standardize class as it looked it was doing standardization. If I am wrong, let me know and I'll adjust it to use sklearn's StandardScaler.  In that case, should I make a class that does this and put it in preprocessing.py?

dvanaken commented on server/website/website/tasks/periodic_tasks.py:
```diff
@@ -9,14 +9,14 @@
 from website.utils import DataUtil, JSONUtil
 
 
-# Log debug messages 
```
Delete space at end of line

dvanaken commented on server/website/website/tasks/periodic_tasks.py:
```diff
@@ -25,15 +25,15 @@
 def run_background_tasks():
     ## 1. Create a new entry in the PipelineRun table for this
     ##    background task
-    
```
Delete space at end of line

dvanaken commented on server/website/website/tasks/periodic_tasks.py:
```diff
@@ -47,7 +47,7 @@ def run_background_tasks():
     for workload in unique_workloads:
         ## 2a. Call first subtask to aggregates the knob & metric data for
         ##     this workload. Save the knob & metric data as (separate)
-        ##     entries in the PipelineData table 
```
Delete space at end of line

dvanaken commented on server/website/website/tasks/periodic_tasks.py:
```diff
@@ -156,7 +156,7 @@ def aggregate_data(workload):
     metric_data = {
         'data': aggregated_data['y_matrix'],
         'rowlabels': copy.deepcopy(aggregated_data['rowlabels']),
-        'columnlabels': aggregated_data['y_columnlabels'] 
```
Delete space at end of line

dvanaken commented on server/website/website/tasks/periodic_tasks.py:
```diff
@@ -187,7 +187,7 @@ def run_workload_characterization(metric_data):
             nonconst_columnlabels.append(cl)
     nonconst_matrix = np.hstack(nonconst_matrix)
     n_rows, n_cols = nonconst_matrix.shape
- 
```
Delete space at end of line

dvanaken commented on server/website/website/tasks/periodic_tasks.py:
```diff
@@ -214,10 +214,10 @@ def run_workload_characterization(metric_data):
     detk = create_kselection_model("det-k")
     detk.fit(components, kmeans_models.cluster_map_)
 
-    # Get pruned metrics, cloest samples of each cluster center 
```
Delete space at end of line

dvanaken commented on server/website/website/tasks/periodic_tasks.py:
```diff
@@ -236,41 +236,42 @@ def run_knob_identification(knob_data, metric_data):
+    nonconst_knob_matrix = []
+    nonconst_knob_columnlabels = []
+
+    for col, cl in zip(knob_matrix.T, knob_columnlabels):
```
Preprocessing comment

dvanaken commented on server/website/website/tasks/periodic_tasks.py:
```diff
@@ -236,41 +236,42 @@ def run_knob_identification(knob_data, metric_data):
+    nonconst_metric_matrix = []
+    nonconst_metric_columnlabels = []
+
+    for col, cl in zip(knob_matrix.T, metric_columnlabels):
```
Preprocessing comment

dvanaken commented on server/website/website/tasks/periodic_tasks.py:
```diff
@@ -236,41 +236,42 @@ def run_knob_identification(knob_data, metric_data):
+            nonconst_metric_columnlabels.append(cl)
+    nonconst_metric_matrix = np.hstack(nonconst_metric_matrix)
+
+    standardizer = Standardize(axis = 0)
```
Preprocessing comment

dvanaken commented on server/website/website/tasks/periodic_tasks.py:
```diff
@@ -236,41 +236,42 @@ def run_knob_identification(knob_data, metric_data):
+    standardized_knob_matrix = standardizer.fit_transform(nonconst_knob_matrix)
+    standardized_metric_matrix = standardizer.fit_transform(nonconst_metric_matrix)
+
+    shuffle_indices = get_shuffle_indices(standardized_knob_matrix.shape[0], seed=17)
```
Preprocessing comment

swimj commented:
Ok, I switched out the scaling to use StandardScaler and added some comments.

Regarding the whitespace, I'm not exactly sure what is causing that.  And I think what actually happened is that there were spaces at the end of some lines in the original code, and my editor (Atom) must have gotten rid of them by default. Since when I look at the diff between my commit and the original, the lines highlighted red are the ones with the extra spaces. Do you mean I should go back and put the spaces in? 

In the future, I think there is a setting I can turn off so that trailing spaces in any line do not get trimmed. Should I do that?

dvanaken commented:
@swimj LGTM. Once you update your branch with the new changes I will merge.

swimj commented:
Sorry, I'm still new to this. What do I do to update my branch?

On Thu, Dec 14, 2017 at 1:55 PM, Dana Van Aken <notifications@github.com>
wrote:

> @swimj <https://github.com/swimj> LGTM. Once you update your branch with
> the new changes I will merge.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cmu-db/ottertune/pull/33#issuecomment-351803074>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AUP6IcHRueXQSnr0-WH_dFqmidQYE4_6ks5tAW8wgaJpZM4RAAAE>
> .
>

dvanaken commented:
No worries! I was actually able to merge it in manually.
